mock/py/mockbuild/package_manager.py

```diff
@@ -353,8 +353,10 @@ def expand_url_vars(self, string):
                 expand[key] = self.config['dnf_vars'][key]
 
         for key, value in expand.items():
-            key = "$" + key
-            string = string.replace(key, value)
+            pattern = r"\$(?:" + re.escape(key)
+            pattern += r"|\{" + re.escape(key) + r"\})"
```

Comment:
Can not we simply
```python
string = string.replace("$" + key, value)
string = string.replace("${}".format(key), value)
```
?

I don't think it is correct to compile patterns in loop.  The re pattern is a bit
harder to read, e.g. `'\\$(?:releasever|\\{releasever\\})'`.

Reply:
Agreed, I think the second line should be

```
string = string.replace("${" + key + "}", value)
```

which is actually just about what I'm doing in my workaround hack.